amplify/backend/function/fetchMotionState/src/index.js

```diff
@@ -3,37 +3,91 @@ const {
   getLatestMotionState,
   updateStakerRewardsInDB,
   getMotionData,
+  didMotionPass,
+  updateMotionMessagesInDB,
 } = require('./utils');
-
 /**
  * @type {import('@types/aws-lambda').APIGatewayProxyHandler}
  */
 exports.handler = async (event) => {
   const { colonyAddress, transactionHash } = event.arguments?.input || {};
   /* Get latest motion state from chain */
   const motionData = await getMotionData(transactionHash);
-  const motionState = await getLatestMotionState(colonyAddress, motionData);
+  const motionState =
+    motionData && (await getLatestMotionState(colonyAddress, motionData));
   /*
    * Check if we need to update staker rewards
    * This ensures the rewards are present in the event a motion fails before going to a vote,
    * so the side that has objected can reclaim their stake.
    */
-  if (transactionHash && motionState === MotionState.Failed) {
-    if (motionData) {
-      const {
-        motionStakes: {
-          percentage: { yay: yayPercent },
-        },
-      } = motionData;
+  if (transactionHash && motionData && motionState === MotionState.Failed) {
```

Comment:
The query should also be skipped if there's no transactionHash, so I'd say the same as above, if you want to be cautious, add the check for a tx hash in an `if` statement at the top of the function. But you'll get a lamda error if there's no tx hash or if it's an empty string:
> ValidationException: One or more parameter values are not valid. The AttributeValue for a key attribute cannot contain an empty string value. Key: id

So you'll know for sure the query is being called incorrectly.

Reply:
Previously, because there was only one side effect in this lamda, adding the tx hash was effectively "putting it at the top" of the function. But now we have a few, I think it should be extracted